src/tools/w3dviewer/datatreeview.cpp

```diff
@@ -639,33 +639,176 @@ void CDataTreeView::RestrictAnims(bool enable)
     }
 }
 
-#if 0
-void CDataTreeView::AddAnimationsForItem(HTREEITEM item)
+void CDataTreeView::RenameItem(const char *newname, const char *oldname, AssetType type)
 {
-    // do later
+    if (newname != nullptr && oldname != nullptr) {
+        SetRedraw(false);
+        HTREEITEM item;
+
+        switch (type) {
+            case ASSET_TYPE_MESH:
+                item = this->m_categoryTreeItems[CATEGORY_MESH];
+                break;
+            case ASSET_TYPE_AGGREGATE:
+                item = this->m_categoryTreeItems[CATEGORY_AGGREGATE];
+                break;
+            case ASSET_TYPE_HLOD:
+                item = this->m_categoryTreeItems[CATEGORY_HLOD];
+                break;
+            case ASSET_TYPE_PARTICLEEMITTER:
+                item = this->m_categoryTreeItems[CATEGORY_EMITTER];
+                break;
+            case ASSET_TYPE_PRIMITIVE:
+                item = this->m_categoryTreeItems[CATEGORY_PRIMITIVES];
+                break;
+            case ASSET_TYPE_SOUND:
+                item = this->m_categoryTreeItems[CATEGORY_SOUNDS];
+                break;
+            default:
+                break;
+        }
+
+        HTREEITEM newitem = FindItemByName(item, oldname);

```

Comment:
This does not look right. It passes uninitialized `HTREEITEM item` here if type is `CATEGORY_MATERIAL`, `CATEGORY_COLLECTION` or `CATEGORY_HIERARCHY`.

`HTREEITEM item = nullptr` perhaps?

Reply:
fixed.